File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/repo/RoleRepository.java
package pers.acp.admin.oauth.repo;

import pers.acp.admin.oauth.base.OauthBaseRepository;
import pers.acp.admin.oauth.entity.Role;

import java.util.List;

/**
 * @author zhangbin by 2018-1-17 17:48
 * @since JDK 11
 */
public interface RoleRepository extends OauthBaseRepository<Role, String> {

    List<Role> findAllByOrderBySortAsc();

    List<Role> findByAppidOrderBySortAsc(String appId);

    void deleteByIdIn(List<String> idList);

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/repo/RoleRepository.kt
package pers.acp.admin.oauth.repo

import pers.acp.admin.common.base.BaseRepository
import pers.acp.admin.oauth.entity.Role

/**
 * @author zhangbin by 2018-1-17 17:48
 * @since JDK 11
 */
interface RoleRepository : BaseRepository<Role, String> {

    fun findAllByCodeOrderBySortAsc(code: String): MutableList<Role>

    fun findAllByOrderBySortAsc(): MutableList<Role>

    fun findByAppIdOrderBySortAsc(appId: String): MutableList<Role>

    fun findByAppIdAndLevelsGreaterThanOrderBySortAsc(appId: String, level: Int): MutableList<Role>

    fun deleteByIdIn(idList: MutableList<String>)

}
